### PR TITLE
Migrates wallets/connections to the use-solana library

### PR DESCRIPTION
**Pull request by macalinao**
This library supports several wallets automatically.

The library has few docs right now-- please let me know ahead of time if you want to merge this

### CODE REVIEW

File: src/contexts/connection.tsx [hunk 1]
```diff
@@ -100,99 +58,50 @@ export function ConnectionProvider({ children = undefined as any }) {
...
+      });
 
       setTokenMap(knownMints);
       setTokens(list);
     })();
   }, [connection, chain]);
 
-  setProgramIds(env);
-
-  // The websocket library solana/web3.js uses closes its websocket connection when the subscription list
```
Comment (Arrowana) [inline src/contexts/connection.tsx:129]: By the way is this supposed hack still required?

**Comment by Arrowana** [comment]
I wanted to see this for a while:
- @saberhq/use-solana Is the package name final? I would like to see it under @solana even if community maintained, just because this is the first foot into dapp and I would like to see the dapp abstraction as building blocks "owned" by solana-labs
- use-solana the name is a bit too broad, can we possibly have something that make it clear it is a wallet abstraction + connection context? Other abstraction should go in another packages imho

**Comment by bartosz-lipinski** [comment]
> I wanted to see this for a while:
> 
> * @saberhq/use-solana Is the package name final? I would like to see it under @Solana even if community maintained, just because this is the first foot into dapp and I would like to see the dapp abstraction as building blocks "owned" by solana-labs
> * use-solana the name is a bit too broad, can we possibly have something that make it clear it is a wallet abstraction + connection context? Other abstraction should go in another packages imho

@macalinao I agree with this, is it worth moving that repo to https://github.com/solana-labs/wallet-provider and we can publish @solana/... from there.

**Comment by macalinao** [comment]
@bartosz-lipinski yes I'd love to have it maintained under the solana org! Let me know how you'd like me to migrate things over

**Pull request closed without merging**